api/listener.go

```diff
@@ -136,8 +140,10 @@ func newIDGenerator(length uint8) *randID {
 	return &randID{length: length}
 }
 
-func (id *randID) newID() string {
+func (id *randID) newID() (string, error) {
 	token := make([]byte, id.length)
-	rand.Read(token)
-	return "0x" + hex.EncodeToString(token)
+	if _, err := rand.Read(token); err != nil {
```

Comment:
Read generates len(p) random bytes from the default Source and writes them into p. It always returns len(p) and a nil error.  Read, unlike the Rand.Read method, is safe for concurrent use.

Reply:
ok, revert